packages/vats/tools/ibc-utils.js

```diff
@@ -0,0 +1,84 @@
+export const REMOTE_ADDR_RE =
+  /^(\/ibc-hop\/[^/]+)*\/ibc-port\/([^/]+)\/(ordered|unordered)\/([^/]+)$/s;
+harden(REMOTE_ADDR_RE);
+/** @typedef {`/ibc-hop/${string}/ibc-port/${string}/{'ordered'|'unordered'}/${string}`} RemoteIbcAddress */
```

Comment:
This is a valiant attempt at expressing the type of an IBC multiaddr, whose basic format is described in [`agoric-sdk/packages/network/src/multiaddr.js`](https://github.com/Agoric/agoric-sdk/tree/master/packages/network/src/multiaddr.js).

The `REMOTE_ADDR_RE` correctly specifies a remote address, but unfortunately, `RemoteIbcAddress` looks like it disallows certain details of the RE's structure, namely:
1. zero or more `/ibc-hop/${nonSlashString}` prefixes, and
2. all but the final `${string}` cannot contain a slash.

Also, there's a simple typo (missing `$` in `${'ordered'|'unordered'}`).